### ISSUE TITLE
Can I block an address in outgoing mail?

### ISSUE THREAD

**Issue by lwl12** [mailcow/mailcow-dockerized#816]
I want to stop sending any mail to some addresses.

**Comment by mkuron** [comment]
See e.g. https://serverfault.com/questions/534174/how-do-i-block-an-email-address-in-postfix and https://serverfault.com/questions/412638/block-outgoing-mail-to-specific-address-using-postfix . I wouldn't recommend the `transport_maps` solution as that may interfere with existing Mailcow functionality, but the others should work easily.

A minimally-invasive alternative that only requires changes to one file would be to put the following into data/conf/postfix/main.cf:
```
recipient_canonical_maps = inline:{ bad.recipient@example.com=rejected@nowhere.invalid }
recipient_canonical_classes = envelope_recipient
```

**Comment by lwl12** [comment]
Is it possible to use wildcards (block entire domain)?
I want to block mail sent to the ten minute mailbox.

**Comment by mkuron** [comment]
Sure, just leave the left side of the _@_ blank.

**Comment by lwl12** [comment]
Thanks for help!

**Comment by lwl12** [comment]
This way seems to be invalid after the update.

```
warning: /opt/postfix/conf/main.cf, line 115: overriding earlier entry: recipient_canonical_maps=inline:{ bad.recipient@example.com=rejected@nowhere.invalid }
```

Also, is there any way to block such addresses as `@*.example.com`?

**Comment by mkuron** [comment]
`recipient_canonical_maps` can now be configured from the admin web UI on the address rewriting tab. Domain wildcards are not supported, but that is theoretically possible by modifying the SQL slightly (see #953 for a list of files that this relates to). Pull requests welcome.

**Comment by PaoloVi** [comment]
recipient_canonical_maps works only for domain.
Is there a way to block only one account to send mail, but still receive?
I tried also with transport_maps, but it was ignored. Probably my fail. This is how I created the transport maps:

docker-compose exec postfix-mailcow postconf -e "transport_maps = hash:/opt/postfix/conf/transport"
docker-compose exec postfix-mailcow postconf -c /opt/postfix/conf transport_maps
docker-compose exec postfix-mailcow touch /opt/postfix/conf/transport
docker-compose exec postfix-mailcow sh -c 'echo "mail@example.com    error (or reject): Send mail from this account disabled" >> /opt/postfix/conf/transport'
docker-compose exec postfix-mailcow cat /opt/postfix/conf/transport
docker-compose exec postfix-mailcow postmap /opt/postfix/conf/transport
docker-compose restart postfix-mailcow
